src/lib/menu/menu-directive.ts

```diff
@@ -196,15 +196,23 @@ export class MatMenu implements AfterContentInit, MatMenuPanel<MatMenuItem>, OnI
    */
   @Input('class')
   set panelClass(classes: string) {
+    if (this.previousPanelClass && this.previousPanelClass.length) {
+      this.previousPanelClass.split(' ').forEach((className: string) => {
+        delete this._classList[className];
```

Comment:
Can you set it to `false` rather than deleting it? We want to avoid changing the shape of the object as much as possible.